lib/src/simple_op_store.rs

```diff
@@ -84,281 +203,518 @@ fn not_found_to_store_error(err: std::io::Error) -> OpStoreError {
+    }
+}
+
+impl OpStore for ThriftOpStore {
+    fn read_view(&self, id: &ViewId) -> OpStoreResult<View> {
+        let path = self.view_path(id);
+        let mut file = File::open(path).map_err(not_found_to_store_error)?;
+        let thrift_view = read_thrift(&mut file)?;
+        Ok(View::from(&thrift_view))
+    }
 
+    fn write_view(&self, view: &View) -> OpStoreResult<ViewId> {
+        let id = hash_view(view);
+        let temp_file = NamedTempFile::new_in(&self.path)?;
+        let thrift_view = simple_op_store_model::View::from(view);
+        write_thrift(&thrift_view, &mut temp_file.as_file())?;
         persist_content_addressed_temp_file(temp_file, self.view_path(&id))?;
         Ok(id)
     }
 
     fn read_operation(&self, id: &OperationId) -> OpStoreResult<Operation> {
         let path = self.operation_path(id);
         let mut file = File::open(path).map_err(not_found_to_store_error)?;
-
-        let proto: crate::protos::op_store::Operation = Message::parse_from_reader(&mut file)?;
-        Ok(operation_from_proto(&proto))
+        let thrift_operation = read_thrift(&mut file)?;
+        Ok(Operation::from(&thrift_operation))
     }
 
     fn write_operation(&self, operation: &Operation) -> OpStoreResult<OperationId> {
+        let id = hash_operation(operation);
         let temp_file = NamedTempFile::new_in(&self.path)?;
+        let thrift_operation = simple_op_store_model::Operation::from(operation);
+        write_thrift(&thrift_operation, &mut temp_file.as_file())?;
+        persist_content_addressed_temp_file(temp_file, self.operation_path(&id))?;
+        Ok(id)
+    }
+}
+
+fn hash_ref_target(ref_target: &RefTarget) -> Vec<u8> {
+    let mut hasher = Blake2b512::new();
+    match ref_target {
+        RefTarget::Normal(id) => {
+            hasher.update(b"0");
+            hasher.update(id.as_bytes());
+        }
+        RefTarget::Conflict { removes, adds } => {
+            hasher.update(b"1");
+            for id in removes {
+                hasher.update(b"0");
+                hasher.update(id.as_bytes());
+            }
+            for id in adds {
+                hasher.update(b"1");
+                hasher.update(id.as_bytes());
+            }
+        }
+    }
+    hasher.finalize().to_vec()
+}
 
-        let proto = operation_to_proto(operation);
-        let mut proto_bytes: Vec<u8> = Vec::new();
-        proto.write_to_writer(&mut proto_bytes)?;
+fn hash_branch_target(branch_target: &BranchTarget) -> Vec<u8> {
+    let mut hasher = Blake2b512::new();
+    match &branch_target.local_target {
+        None => {
+            hasher.update(b"0");
+        }
+        Some(ref_target) => {
+            hasher.update(b"1");
+            hasher.update(&hash_ref_target(ref_target));
+        }
+    }
+    for (name, ref_target) in branch_target
+        .remote_targets
+        .iter()
+        .sorted_by_key(|(name, _)| *name)
+    {
+        hasher.update(name.as_bytes());
+        hasher.update(b"\0");
```

Comment:
Do we have any assurances that `name` doesn't contain nulls? It might be more robust to tag with the length of `name` instead (and similarly below).